Frameworks/QuartzCore/CALayer.mm

```diff
@@ -1570,9 +1568,9 @@ - (CATransform3D)sublayerTransform {
 */
 - (void)setBackgroundColor:(CGColorRef)color {
     if (color != nil) {
-        priv->backgroundColor = *[static_cast<UIColor*>(color) _getColors];
+        priv->backgroundColor = *(reinterpret_cast<const __CGColorQuad*>(CGColorGetComponents(color)));
```

Comment:
This is with the knowledge of transparent information. 
the backing of CGColorGetComponents are a quad struct. Another way is to have a private getter or just take it out of the Quad struct and put it into another Quad struct. 

---
In reply to: [113286126](https://github.com/Microsoft/WinObjC/pull/2538#discussion_r113286126) [](ancestors = 113286126)